### ISSUE TITLE
Request cancellation via OpenAI API does not seem to work

### ISSUE THREAD

**Issue by KizzyCode** [LostRuins/koboldcpp#745]
If I use koboldcpp's OpenAI API via SillyTavern or LibreChat, and then cancel the request via the stop buttons, koboldcpp happily keeps generating new tokens until either the token limit is reached or it comes to a conclusion.

I'm not 100% sure if that's a problem with koboldcpp; but since both frontends seem to work with other backends and fail with koboldcpp, I'd guess it is the outlier here.

Steps to reproduce:
1. Setup SillyTavern or LibreChat
2. Connect it to koboldcpp via the OpenAI v1 API
3. Try to abort a request

**Comment by LostRuins** [comment]
Are you using streaming?

**Comment by KizzyCode** [comment]
Positive, I'm using streaming. Also, in like 1 out of 4 cases it seems to abort correctly, so I'm a bit puzzled... 
This is my koboldcpp config: [miqu-1-70b.q5_K_M.kcpps.zip](https://github.com/LostRuins/koboldcpp/files/14595419/miqu-1-70b.q5_K_M.kcpps.zip). I don't know of a good way to export SillyTavern's or LibreChat's config, but both are pretty vanilla.

**Comment by LostRuins** [comment]
What version of sillytavern and koboldcpp are you using? Did you select the "koboldCpp" option under text-completions endpoints?

**Comment by KizzyCode** [comment]
- koboldcpp: KoboldCPP-v1.60.1.yr0-ROCm
- SillyTavern: SillyTavern 1.11.5 'release' (ad36b3b6)

Api-type is Text Completion/KoboldCpp: 
![Screenshot 2024-03-14 at 15 15 13](https://github.com/LostRuins/koboldcpp/assets/13999009/14d417d0-e2d1-4d32-bcd7-d405bcd092ae)

**Comment by LostRuins** [comment]
Hmm that is odd then. I'm not very sure, but i'll look into it. Did you try see if it mainline koboldcpp works compared to the rocm fork?

**Comment by KizzyCode** [comment]
Nope, I didn't; but I can try later, just to be sure it's not related to the rocm patches. Will give an update :)
